docs/source/release_notes.rst

```diff
@@ -2,6 +2,7 @@ Release Notes
 -------------
 **Future Release**
     * Enhancements
+        * Added `DatetimeFormatDataCheck` for time series problems :pr:`2603`
```

Comment:
You need two `'s to format this as code, one italicizes!

https://feature-labs-inc-evalml--2603.com.readthedocs.build/en/2603/release_notes.html